src/Ulid.Unity/Assets/Scripts/Ulid/Ulid.cs

```diff
@@ -155,21 +183,58 @@ internal Ulid(ReadOnlySpan<char> base32)
             randomness8 = (byte)((CharToBase32[base32[22]] << 7) | (CharToBase32[base32[23]] << 2) | (CharToBase32[base32[24]] >> 3));
         }
 
+#if NETCOREAPP3_0_OR_GREATER
+        private static readonly Vector128<byte> guidVecShuffle = Vector128.Create((byte)3, 2, 1, 0, 5, 4, 7, 6, 8, 9, 10, 11, 12, 13, 14, 15);
```

Comment:
For newer targets (.NET 6.0 onwards) this constant should be given "inline" for allowing the JIT to emit better code. As at the moment only .NET 6+ is officially supported I wouldn't bother with .NET Core 3.1 optimizations anymore. So the below `#if` def can be dropped. For the `Shuffle` you could create a private method that handles this depending on the target.